### ISSUE TITLE
Advice in UnitsWarning reading FITS file does not work

### ISSUE THREAD

**Issue by taldcroft** [astropy/astropy#14330]
### Description

I'm reading a FITS file with field that has units of `DN`. On doing so I get this warning:
```
chandra_aca/tests/test_maude_decom.py::test_vcdu_vs_level0
  /Users/aldcroft/miniconda3/envs/ska3-flight-2023.1rc4/lib/python3.10/site-packages/astropy/units/core.py:2097: 
UnitsWarning: 'DN' did not parse as fits unit: At col 0, Unit 'DN' not supported by the FITS standard. 
Did you mean dN or daN? If this is meant to be a custom unit, define it with 'u.def_unit'. 
To have it recognized inside a file reader or other code, enable it with 'u.add_enabled_units'. 
For details, see https://docs.astropy.org/en/latest/units/combining_and_defining.html
    warnings.warn(msg, UnitsWarning)
```

However, adding `u.add_enabled_units([u.DN])` fails to make the warning go away.

### Expected behavior

When I follow the instructions in the warning I expect the warning to disappear.

### Actual behavior

I still got the warning even with `u.DN` enabled as a unit.

### Steps to Reproduce

If necessary to resolve this I can provide a reproducible example.

### System Details

```
>>> import platform; print(platform.platform())
macOS-13.0.1-x86_64-i386-64bit
>>> import sys; print("Python", sys.version)
Python 3.10.8 | packaged by conda-forge | (main, Nov 22 2022, 08:27:35) [Clang 14.0.6 ]
>>> import numpy; print("Numpy", numpy.__version__)
Numpy 1.23.5
>>> import erfa; print("pyerfa", erfa.__version__)
pyerfa 2.0.0.1
>>> import astropy; print("astropy", astropy.__version__)
astropy 5.2.1
>>> import scipy; print("Scipy", scipy.__version__)
Scipy 1.10.0
>>> import matplotlib; print("Matplotlib", matplotlib.__version__)
Matplotlib 3.6.3
```

**Comment by pllim** [comment]
DN was added by @jdavies-st in https://github.com/astropy/astropy/pull/11591 so hopefully he can comment. 🤞

**Comment by jdavies-st** [comment]
Well, DN is probably the most common unit used across astronomy for counts that is not an "official" FITS unit.

But DN is a unit allowed by astropy:

https://github.com/astropy/astropy/blob/a0885a98841419c7bf9249fc9f75207b5259a18f/astropy/units/astrophys.py#L179-L183

But it is not listed in the official FITS subclass of units, which is quite strict about warning the user

https://github.com/astropy/astropy/blob/7539d76ceae146f930d4473107d9940d2fc0b74f/astropy/units/format/fits.py#L81-L84

which gets handled here

https://github.com/astropy/astropy/blob/a0885a98841419c7bf9249fc9f75207b5259a18f/astropy/units/core.py#L2071-L2093

Perhaps setting `parse_strict="silent"` would work?  Current in the FITS readers it is set to "warn":

https://github.com/astropy/astropy/blob/0f0949cbd07ab09bae7f95227a658468d1c8c442/astropy/io/fits/convenience.py#L611-L613

I don't think it is useful to warn the user about "DN" since it is so widespread, even though not technically FITS standard - used by everything, including HST, JWST, Spitzer, etc.

It is a separate issue that the warning (or error) message generated only knows about what is in the "fits" format unit, not that the unit is just fine outside the "fits" format.

**Comment by taldcroft** [comment]
OK good, reading with `unit_parse_strict="silent"` was the right thing to silence that warning. 

But this is option is really under the radar. Looking at https://docs.astropy.org/en/latest/search.html?q=unit_parse_strict&check_keywords=yes&area=default, the only mentions in the entire astropy docs are within `timeseries`. I only found the correct option using `Table.read.help(format='fits')`.

Some actions from this could be:
- [ ] Update the warning message to advise setting `unit_parse_strict="silent"`. The current message talks about enabling the unit, but that doesn't seem to be the real problem.
- [ ] Put a section to this effect in the narrative docs, advising on how to deal with non-FITS units.
- [ ] Figure out why `unit_parse_strict` is not apparent in any API docs.

**Comment by mhvk** [comment]
Hmm, the message was supposed to give the right solution! With `parse_strict="silent"` do you actually get the proper DN unit, or an `UnrecognizedUnit`?

I agree that we should be a bit more liberal on input, and recognize common "mistakes". With no warning on input, but perhaps a warning on output (which could be unit specific, i.e., give an alternative if it exists).

**Comment by saimn** [comment]
Also related to other issues/PR e.g. #11842, #12808, #11851, and probably others ;)

I think there are two main issues:
- what to do with units that are commonly used in FITS files but not present in the FITS standard, e.g. `electron` and variants (`e`), lowercase vs uppercase, e.g. `adu` is valid but `ADU` is not, 'DN' etc. 
- even if we add some special cases for the most common ones, there will always be unrecognized units. So how can we tell the readers to recognize some unit ? We had some discussions about using the `generic` format, maybe that was not the best solution (#12808) but then we need a better solution :)

**Comment by mhvk** [comment]
I thought we had solved the second point, of allowing users to tell what unit is meant, with the custom unit snippet that the warning generates. If that doesn't work, that's a bug!

The more general mistake recognition is indeed somewhat separate from that - and can be added separately (maybe with a new `parse_units="liberal"` option or so...)

**Comment by jdavies-st** [comment]
A solution would be, when reading FITS files with units, to use any unit in astropy, not just those allowed via FITS standard.  Feel free to warn on write when using non-FITS standard units, but on read, be silent if parsing can figure it out.

**Comment by mhvk** [comment]
Yes, that is a reasonable option. Indeed, I could have sworn there was already a PR for it, but I cannot seem to find it...

Note that, independently, we should make sure the tip about how to enable units actually works!

**Comment by maxnoe** [comment]
The PR was about `Table.read`: https://github.com/astropy/astropy/pull/11843

**Comment by maxnoe** [comment]
Maybe `parse_strict` could gain a `False` or `"off"` option would disable format-specific restrictions and try to parse it using the full astropy string parser?

**Comment by maxnoe** [comment]
>  Figure out why unit_parse_strict is not apparent in any API docs.

That's a problem I have had often before: how to figure out which format specific options can be passed to `Table.read`
